src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/factory/DesignPattern.java

```diff
@@ -0,0 +1,94 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.factory;
+
+public class DesignPattern implements Refactorable {
+
+    private String name;
+    private String patternCode;
+    private String example;
+    private DesignPatternType type;
+
+    private DesignPattern(Builder builder) {
+        this(builder.name, builder.patternCode, builder.example, builder.type);
+    }
+
+    private DesignPattern(String name, String patternCode, String example, DesignPatternType type) {
+        this.name = name;
+        this.patternCode = patternCode;
+        this.example = example;
+        this.type = type;
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public String getPatternCode() {
+        return this.patternCode;
+    }
+
+    public String getExample() {
+        return this.example;
+    }
+
+    public DesignPatternType getType() {
+        return this.type;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public void setPatternCode(String patternCode) {
+        this.patternCode = patternCode;
+    }
+
+    public void setExample(String example) {
+        this.example = example;
+    }
+
+    static Builder builder() {
+        return new Builder();
+    }
+
+    public boolean isSameType(DesignPattern designPattern) {
+        return type.equals(designPattern.type);
+    }
+
+    @Override
+    public String refactor(String badCode) {
```

Comment:
Refactorable interface doesn't make sense at all with builder and non-inheritance approach. Point of Refactorable is to deliver different behaviour with different subclasses. Now, we assume magic of refactor will be a result of interaction with private fields.

Reply:
do not agree (at least not in 100%),  please see:

>  "some magic involving **patternCode** happens here"